internal/loader/typed_setting.go

```diff
@@ -118,3 +104,51 @@ func isJsonContentType(contentType *string) bool {
 	matched, _ := regexp.MatchString("^application\\/(?:[^\\/]+\\+)?json(;.*)?$", contentTypeStr)
 	return matched
 }
+
+// UnmarshalConfigMap is used for scenarios related to feature flag refresh.
+// Currently, only supports feature flag settings in JSON/YAML formats.
+func UnmarshalConfigMap(existingConfigMapSetting *map[string]string, dataOptions *acpv1.ConfigMapDataOptions) (map[string]interface{}, map[string]interface{}, error) {
+	allSettings := make(map[string]interface{})
+	existingConfigMapData := (*existingConfigMapSetting)[dataOptions.Key]
+	var typedFeatureFlags map[string]interface{}
+	var err error
+
+	switch dataOptions.Type {
```

Comment:
We should safe guard that the type should be either `json` or `yaml`. Could we throw exception here when data type is not `json` or `yaml`?